kalite/distributed/static/js/distributed/user/views.js

```diff
@@ -0,0 +1,285 @@
+        }
+        if (this.start_open) {
+            this.show_modal();
+        }
+    },
+
+    close_modal: function() {
+        $("#loginModal").modal('hide');
+    },
+
+    show_modal: function() {
+        $("#loginModal").modal('show');
+    }
+});
+
+window.LoginView = BaseView.extend({
+
+    events: {
+        "click .login-btn": "login",
+        "click .password-btn": "toggle_password",
+        "change #id_facility": "facility_change",
+        "keypress #id_username": "key_user",
+        "keypress #id_password": "key_pass"
+    },
+
+    template: HB.template("user/login"),
+
+    initialize: function() {
+        _.bindAll(this);
+        this.facility = (this.model.get("facilities")[0] || {id:""}).id;
+        this.admin = false;
+        if (this.model.get("simplified_login")) {
+            this.set_autocomplete();
+        }
+        this.render();
+    },
+
+    render: function() {
+        this.render_data = {
+            "simplified_login": this.model.get("simplified_login")
+        };
+        if (this.model.get("facilities").length > 1) {
+            this.render_data["facilities"] = this.model.get("facilities");
+        }
+        this.$el.html(this.template(this.render_data));
+    },
+
+    login: function() {
+        this.$("input").parent().removeClass("has-error");
+        username = this.$("#id_username").val();
+        password = this.$("#id_password").val();
+        facility = this.facility || this.$("#id_facility").val();
+        if (!username) {
+            this.$("#id_username").parent().addClass("has-error");
+        } else if (!password && (!this.model.get("simplified_login") || this.admin)) {
+            this.$("#id_password").parent().addClass("has-error");
+        } else {
+            this.model.login(username, password, facility, this.handle_login);
+        }
+    },
+
+    handle_login: function(response) {
+        if (response.status == 200) {
+            this.trigger("login_success");
+        } else {
+            var error_data = JSON.parse(response.responseText);
+            this.$("#id_" + error_data.error_highlight).parent().addClass("has-error");
+            if (error_data.error_highlight == "password") {
+                this.$("#id_" + error_data.error_highlight).val("");
+            }
+        }
+    },
+
+    toggle_password: function() {
+        this.admin = !this.admin;
+        if (this.admin) {
+            this.$("#id_username").autocomplete("disable");
+            this.$("#id_password-container").show();
+        } else {
+            this.$("#id_username").autocomplete("enable");
+            this.$("#id_password-container").hide();
+        }
+    },
+
+    facility_change: function(event) {
+        this.facility = this.$("#id_facility").val();
+        if (this.model.get("simplified_login")) {
+            this.set_autocomplete();
+        }
+    },
+
+    set_autocomplete: function() {
```

Comment:
Just make a comment that this initializes `self.student_usernames` for `start_autocomplete`